modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastEffectsViewModel.kt

```diff
@@ -24,12 +26,15 @@ class PodcastEffectsViewModel
 @Inject constructor(
     private val podcastManager: PodcastManager,
     private val playbackManager: PlaybackManager,
-    private val settings: Settings
+    private val settings: Settings,
+    private val analyticsTracker: AnalyticsTrackerWrapper,
 ) : ViewModel(), CoroutineScope {
 
     override val coroutineContext: CoroutineContext
         get() = Dispatchers.Default
 
+    private var isSpeedChanged: Boolean = false
+    private var finalSpeed: Double = 1.0
```

Comment:
Both here and in the `EffectsFragment`, instead of having a default `finalSpeed`, I think we could default it to `null` and use a `null` `finalSpeed` as indicating the speed had not been changed and a non-`null` `finalSpeed` as indicating that the speed had been changed (renaming it to something like `updatedSpeed` might fit better in that case).

This would have the benefit of dropping the `isSpeedChanged` field. Also, it would avoid us defaulting `finalSpeed` to 1.0 even when that isn't the actual starting speed (i.e., when the speed had been changed previously).

I don't feel strongly about this change, so this is more just food for thought. Definitely feel free to leave it as-is if you prefer.

Reply:
Yep, it'll be an improved version this way, I applied it in 74626ec.